js.go

```diff
@@ -2166,6 +2168,19 @@ func (sub *Subscription) Fetch(batch int, opts ...PullOpt) ([]*Msg, error) {
 		defer cancel()
 	}
 
+	deadline, hasDeadline := ctx.Deadline()
+	if !hasDeadline {
+		// Prevent from passing the background context which will just block
+		// and cannot be canceled either.
+		if octx, ok := ctx.(ContextOpt); ok && octx.Context == context.Background() {
+			return nil, ErrNoDeadlineContext
+		}
+		// If the context did not have a deadline, then create a new child context
+		// that will use the default timeout from the JS context.
+		ctx, cancel = context.WithTimeout(ctx, ttl)
```

Comment:
Do we need to create a context with timeout from the user-given context here? If not, then code could be even more refactored to check for `ctx = o.ctx` being not nil first, just to check for "has deadline" and if so set ctx to nil so there is then the check for `nil` and creation with background() context. If we need to use the user-given context, then ok with this code.

Reply:
Yes we need to chain the context from the parent one so that if it is canceled then the child one is canceled as well.